### PR TITLE
cclient: implement iota_api_get_node_info

### PR DESCRIPTION
**Pull request by oopsmonk**
This PR implements iota_api_get_node_info.

# Test Plan:
Unit tests in //cclient/serialization:test_json

### CODE REVIEW

File: cclient/serialization/json/json_serializer.c [hunk 1]
```diff
@@ -40,6 +40,52 @@ retcode_t utarray_to_json_array(UT_array* ut, cJSON* json_root,
   return RC_OK;
 }
 
+retcode_t json_get_size(const cJSON* json_obj, const char* obj_name,
+                        size_t* num) {
+  cJSON* json_value = cJSON_GetObjectItemCaseSensitive(json_obj, obj_name);
+  if (json_value == NULL) return RC_CCLIENT_JSON_KEY;
+
+  if (cJSON_IsNumber(json_value)) {
```
Comment (tsvisabo) [inline cclient/serialization/json/json_serializer.c:48]: Can it be negative? would you return error in that case?

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: cclient/serialization/json/json_serializer.c (hunk 1 above)
Comment (oopsmonk) [reply cclient/serialization/json/json_serializer.c:48]: it's a boolean and return RC_CCLIENT_JSON_PARSE if the value is not a number.

Comment (tsvisabo) [reply cclient/serialization/json/json_serializer.c:48]: Yes, i was referring to: `(size_t)json_value->valuedouble`
could that line be casting negative number to `size_t`?

Comment (oopsmonk) [reply cclient/serialization/json/json_serializer.c:48]: could the value be negative in the response? ex. memory size, time, index of a milestone...

**Review by tsvisabo — APPROVED** [review]

**Pull request merged**